README.md

```diff
@@ -12,3 +12,61 @@ the function and run automatically when you commit through a GitHub action.
 
 What is the span of the parallel program, in terms of worst-case $\Theta$? Hint:
 It may help to consider the DAG of the parallel program.
+
+First, consider my code:
+```js
+function msort(x, lo, hi, tmp, callback) {
+  if (lo >= hi) {
+    callback();
+    return;
+  }
+  var mid = Math.floor((lo + hi) / 2);
+  async.parallel(
+    [
+      (done) => msort(x, lo, mid, tmp, done), // left half
+      (done) => msort(x, mid + 1, hi, tmp, done), // right half
+    ],
+      () => {
+      merge(x, lo, mid, hi, tmp);
+      callback();
+    }
+  );
+}
+function mergesort(x, callback) {
+  const tmp = [];
+  msort(x, 0, x.length - 1, tmp, callback);
+}
+function merge(x, lo, mid, hi, tmp) {
+  let a = lo, b = mid + 1;
+  for (let k = lo; k <= hi; k++) {
+    if (a <= mid && (b > hi || x[a] <= x[b])) {
+      tmp[k] = x[a++];
+    } else {
+      tmp[k] = x[b++];
+    }
+  }
+  for (let k = lo; k <= hi; k++) {
+    x[k] = tmp[k];
+  }
+}
+```
+
+Recall that for vanilla mergesort, the time complexity is $\Theta (n \log n)$. This is because you are breaking the array into $\log n$ sizes of subarrays and going through the array $n$ at each of those array sizes. Hence, you multiple and get $\Theta (n \log n)$. 
+
+My implementation processes every subarray of size $k$ in parallel (provided capability for the parallel processes), so with this implementation, the span is $\log {n} $. Because of the merging process which is $\Theta(n)$ at $\log {n}$ levels, the time complexity is still $\Theta (n \log {n})$. 
```

Comment:
The span assumes that all nodes in the DAG represent constant work -- if they don't, the span would be different.

Reply:
So the span should be $n  \Theta (n)$ because of the merging process which is $\Theta(n)$ at $\log {n}$ levels?